core/src/test/scala/unit/kafka/admin/ReassignPartitionsCommandTest.scala

```diff
@@ -218,8 +202,183 @@ class ReassignPartitionsCommandTest extends ZooKeeperTestHarness with Logging wi
+      (DynamicConfig.Broker.FollowerReplicationThrottledRateProp, "10"),
+      (DynamicConfig.Broker.LeaderReplicationThrottledRateProp, "100"),
+      ("useful.key", "value")
+    )
+
+    //Setup
+    val zk = stubZK(brokers = brokers)
+    val admin = createMock(classOf[AdminUtilities])
+    val propsCapture: Capture[Properties] = newCapture(CaptureType.ALL)
+    expect(admin.fetchEntityConfig(is(zk), is(ConfigType.Topic), anyString())).andStubReturn(new Properties)
+    expect(admin.changeBrokerConfig(is(zk), anyObject().asInstanceOf[Seq[Int]], capture(propsCapture))).anyTimes()
+    //Stub each invocation as EasyMock caches the return value which can be mutated
+    expect(admin.fetchEntityConfig(is(zk), is(ConfigType.Broker), is("100"))).andReturn(copyOf(existingBrokerConfigs))
+    expect(admin.fetchEntityConfig(is(zk), is(ConfigType.Broker), is("101"))).andReturn(copyOf(existingBrokerConfigs))
+    expect(admin.fetchEntityConfig(is(zk), is(ConfigType.Broker), is("102"))).andReturn(copyOf(existingBrokerConfigs))
+    replay(admin)
+
+    //When
+    ReassignPartitionsCommand.removeThrottle(zk, proposed, status, admin)
+
+    //Then props should have gone (dummy remains)
+    for (capture <- propsCapture.getValues)
+      assertEquals("value", capture.get("useful.key"))
+    assertEquals(3, propsCapture.getValues.size()) //3 brokers
+  }
+
+  @Test
+  def shouldRemoveThrottleReplicaListBasedOnProposedAssignment(): Unit = {
+
+    //Given two topics with existing config
+    val proposed = mutable.Map(
+      TopicAndPartition("topic1", 0) -> Seq(100, 101),
+      TopicAndPartition("topic2", 0) -> Seq(100, 101)
+    )
+    val status = mutable.Map(TopicAndPartition("topic1", 0) -> ReassignmentCompleted)
+    val existingConfigs = CoreUtils.propsWith(
+      (LogConfig.LeaderReplicationThrottledReplicasProp, "1:100:2:100"),
+      (LogConfig.FollowerReplicationThrottledReplicasProp, "1:101,2:101"),
+      ("useful.key", "value")
+    )
+
+    //Setup
+    val zk = stubZK(brokers = Seq(100, 101))
+    val admin = createMock(classOf[AdminUtilities])
+    val propsCapture: Capture[Properties] = newCapture(CaptureType.ALL)
+    expect(admin.fetchEntityConfig(is(zk), is(ConfigType.Broker), anyString())).andStubReturn(new Properties)
+    expect(admin.fetchEntityConfig(is(zk), is(ConfigType.Topic), is("topic1"))).andStubReturn(copyOf(existingConfigs))
+    expect(admin.fetchEntityConfig(is(zk), is(ConfigType.Topic), is("topic2"))).andStubReturn(copyOf(existingConfigs))
+
+    //Should change both topics
+    expect(admin.changeTopicConfig(is(zk), is("topic1"), capture(propsCapture)))
+    expect(admin.changeTopicConfig(is(zk), is("topic2"), capture(propsCapture)))
+
+    replay(admin)
+
+    //When
+    ReassignPartitionsCommand.removeThrottle(zk, proposed, status, admin)
+
+    //Then props should have gone (dummy remains)
+    for (actual <- propsCapture.getValues)
+      assertEquals("value", actual.getProperty("useful.key"))
```

Comment:
I think we should check that the properties are null, I broker the code on purpose and it still passed without that.